hera_sim/simulate.py

```diff
@@ -31,226 +39,386 @@ def new_func(*args, **kwargs):
-        if seed != -1:
-            kwargs["seed"] = seed
-
-        # calculate the effect
+        # Simulate the effect by iterating over baselines and polarizations.
         data = self._iteratively_apply(
             model,
             add_vis=add_vis,
             ret_vis=ret_vis,
             vis_filter=vis_filter,
-            antpairpol_cache=antpairpol_cache,
+            antpairpol_cache=self._antpairpol_cache[model_key],
+            seed=seed,
             **kwargs,
-        )
+        )  # This is None if ret_vis is False
 
-        # log the component and its kwargs, if added to data
         if add_vis:
-            # note the filter used if any
-            if vis_filter is not None:
-                kwargs["vis_filter"] = vis_filter
-            # note the defaults used if any
+            # Record the component simulated and the parameters used.
             if defaults._override_defaults:
-                kwargs["defaults"] = defaults()
-            # log the component and the settings used
-            self._components[component] = kwargs
-            # update the history
+                for param in getattr(model, "kwargs", {}):
+                    if param not in kwargs and param in defaults():
+                        kwargs[param] = defaults(param)
             self._update_history(model, **kwargs)
-            # track the seed(s) used, if any
-            if seed != -1:
-                self._update_seeds(self._get_model_name(model))
+            # Record the random state in case no seed was specified.
+            # This ensures that the component can be recovered later.
+            kwargs["seed"] = seed
+            self._update_seeds(model_key)
+            if vis_filter is not None:
+                kwargs["vis_filter"] = vis_filter
+            self._components[model_key] = kwargs
+            self._components[model_key]["alias"] = component
         else:
-            # if we're not adding it, then we don't want to keep
-            # the antpairpol cache
-            _ = self._antpairpol_cache.pop(model_key)
+            del self._antpairpol_cache[model_key]
 
-        # return the data if desired
-        if ret_vis:
-            return data
+        return data
 
-    def get(self, component, ant1=None, ant2=None, pol=None):
-        # TODO: docstring
-        # TODO: figure out if this could be handled by _iteratively_apply
+    def get(
+        self,
+        component: Component,
+        key: Optional[Union[int, str, AntPair, AntPairPol]] = None,
+    ) -> Union[np.ndarray, Dict[int, np.ndarray]]:
         """
+        Retrieve an effect that was previously simulated.
+
+        Parameters
+        ----------
+        component
+            Effect that is to be retrieved. See .. meth:: add for more details.
+        key
+            Key for retrieving simulated effect. Possible choices are as follows:
+                An integer may specify either a single antenna (for per-antenna
```

Comment:
I think Sphinx is going to complain about the extra indent here.

Reply:
Sphinx didn't have any issues with it, and it looks fine on the RTD! Though looking through the RTD made me realize there are some other formatting issues I need to fix.